File: app/src/main/java/com/radionula/radionula/data/db/NulaDatabase.java
package com.radionula.radionula.data.db;

import android.content.ContentValues;
import android.content.Context;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;

import com.radionula.radionula.model.NulaTrack;
import com.radionula.radionula.util.MyDatabaseHelper;

import java.util.ArrayList;
import java.util.List;

public class NulaDatabase {
    private MyDatabaseHelper databaseHelper;
    private SQLiteDatabase database;

    public final static String EMP_TABLE = "NulaTracks"; // name of table

    public final static String EMP_ID = "_id"; // id value for track
    public final static String EMP_ARTIST = "artist";  // artist name of track
    public final static String EMP_TITLE = "title";  // title of track
    public final static String EMP_IMAGE = "image";  // image of track

    /**
     * @param context
     */
    public NulaDatabase(Context context) {
        databaseHelper = new MyDatabaseHelper(context);
        database = databaseHelper.getWritableDatabase();
    }


    public long insertTrack(NulaTrack track) {
        ContentValues values = new ContentValues();
        values.put(EMP_ARTIST, track.getArtist());
        values.put(EMP_TITLE, track.getTitle());
        values.put(EMP_IMAGE, track.getImage());
        return database.insert(EMP_TABLE, null, values);
    }

    public List<NulaTrack> selectAllTracks() {
        String[] cols = new String[]{EMP_ID, EMP_ARTIST, EMP_TITLE, EMP_IMAGE};
        Cursor mCursor = database.query(true, EMP_TABLE, cols, null
                , null, null, null, null, null);

        List<NulaTrack> result = new ArrayList<>();

        if (mCursor != null) {
            if (mCursor.moveToFirst()) {
                do {
                    int id = mCursor.getInt(0);
                    String artist = mCursor.getString(1);
                    String title = mCursor.getString(2);
                    String image = mCursor.getString(3);

                    NulaTrack track = new NulaTrack(artist, title, image, id);
                    result.add(track);
                } while (mCursor.moveToNext());
            }
        }
        return result;
    }

    public long remoteTrack(NulaTrack track){
        return database.delete(EMP_TABLE,"_id=?",new String[]{""+track.getId()});
    }

}

File: app/src/main/java/com/radionula/radionula/data/db/NulaDatabase.kt
package com.radionula.radionula.data.db

import android.content.ContentValues
import android.content.Context
import com.radionula.radionula.model.NulaTrack
import com.radionula.radionula.util.MyDatabaseHelper
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.withContext

/**
 * The favourites store, a plain SQLiteOpenHelper.
 *
 * Every call hops to Dispatchers.IO. Opening the database creates the file on
 * first touch, and that used to happen on the main thread - the helper was
 * constructed eagerly in the constructor, and the list query ran straight from
 * an adapter's initialiser.
 */
class NulaDatabase(context: Context) {

    private val helper = MyDatabaseHelper(context)

    suspend fun insertTrack(track: NulaTrack): Long = withContext(Dispatchers.IO) {
        val values = ContentValues().apply {
            put(COLUMN_ARTIST, track.artist)
            put(COLUMN_TITLE, track.title)
            put(COLUMN_IMAGE, track.image)
        }
        helper.writableDatabase.insert(TABLE, null, values)
    }

    suspend fun selectAllTracks(): List<NulaTrack> = withContext(Dispatchers.IO) {
        helper.readableDatabase.query(
            TABLE,
            arrayOf(COLUMN_ID, COLUMN_ARTIST, COLUMN_TITLE, COLUMN_IMAGE),
            null, null, null, null, null
        ).use { cursor ->
            buildList {
                while (cursor.moveToNext()) {
                    add(
                        NulaTrack(
                            artist = cursor.getString(1),
                            title = cursor.getString(2),
                            image = cursor.getString(3),
                            id = cursor.getInt(0)
                        )
                    )
                }
            }
        }
    }

    suspend fun removeTrack(track: NulaTrack): Int = withContext(Dispatchers.IO) {
        helper.writableDatabase.delete(TABLE, "$COLUMN_ID = ?", arrayOf(track.id.toString()))
    }

    private companion object {
        const val TABLE = "NulaTracks"
        const val COLUMN_ID = "_id"
        const val COLUMN_ARTIST = "artist"
        const val COLUMN_TITLE = "title"
        const val COLUMN_IMAGE = "image"
    }
}
